pkg/resources/object_renaming_lists_and_sets.go

```diff
@@ -0,0 +1,518 @@
+	// meaning max_order == len(list).
+	"manually_ordered_list": {
+		Optional:         true,
+		Type:             schema.TypeList,
+		DiffSuppressFunc: ignoreOrderAfterFirstApplyWithManuallyOrderedList("manually_ordered_list"),
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"order": {
+					Type:     schema.TypeInt,
+					Required: true,
+				},
+				"name": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+			},
+		},
+	},
+	// The ordered_list field was an attempt of making manual work done in manually_ordered_list automatic by making the order field computed.
+	// It didn't work because in DiffSuppressFunc it's hard to get the computed value in the "after" state to compare against.
+	// Possibly (but with very low probability), the solution could work by introducing a Computed + Optional list that would be managed by a custom diff function.
+	// Due to increased complexity, it was left as is and more research was dedicated to manually_ordered_list.
+	"ordered_list": {
+		Optional:         true,
+		Type:             schema.TypeList,
+		DiffSuppressFunc: ignoreOrderAfterFirstApplyWithOrderedList("ordered_list"),
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"order": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	},
+}
+
+func ObjectRenamingListsAndSets() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateObjectRenamingListsAndSets,
+		UpdateContext: UpdateObjectRenamingListsAndSets,
+		ReadContext:   ReadObjectRenamingListsAndSets,
+		DeleteContext: DeleteObjectRenamingListsAndSets,
+
+		Schema: objectRenamingListsAndSetsSchema,
+	}
+}
+
+func CreateObjectRenamingListsAndSets(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	ObjectRenamingDatabaseInstance.List = collections.Map(d.Get("list").([]any), func(item any) objectRenamingDatabaseListItem {
+		var name string
+		if nameValue, ok := item.(map[string]any)["name"]; ok {
+			name = nameValue.(string)
+		}
+		return objectRenamingDatabaseListItem{
+			Name:   name,
+			String: item.(map[string]any)["string"].(string),
+			Int:    item.(map[string]any)["int"].(int),
+		}
+	})
+
+	ObjectRenamingDatabaseInstance.OrderedList = objectRenamingDatabaseOrderedListFromSchema(d.Get("ordered_list").([]any))
+	ObjectRenamingDatabaseInstance.ManuallyOrderedList = objectRenamingDatabaseManuallyOrderedListFromSchema(d.Get("manually_ordered_list").([]any))
+
+	d.SetId("identifier")
+
+	return ReadObjectRenamingListsAndSets(ctx, d, meta)
+}
+
+func UpdateObjectRenamingListsAndSets(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	if d.HasChange("list") {
+		// It wasn't working with d.getChange(). It was returning null elements in one of the test case's steps.
+		oldList := d.GetRawState().AsValueMap()["list"].AsValueSlice()
```

Comment:
question: can we get nil in any of these two (at the `["list"]`)? If so, we should handle it.